File: CS/Models/GridViewSettingsHelper.cs
using System.Web.Mvc;
using System.Web.UI;
using DevExpress.Web.Mvc;
using DevExpress.Web.Mvc.UI;

public static class GridViewSettingsHelper {
    public static GridViewSettings CreateGridViewSettings(this WebViewPage view, GridType gridType) {
        GridViewSettings settings = new GridViewSettings();
        settings.Name = "GridView" + gridType.ToString();
        settings.CallbackRouteValues = new { Controller = "Home", Action = "GridViewPartial", gridType = gridType };
        settings.KeyFieldName = "Id";

        if (gridType == GridType.Invoices) {
            settings.Columns.Add("Id");
            settings.Columns.Add("Price");
            settings.Columns.Add(column => {
                column.Caption = "Description";
                column.SetDataItemTemplateContent(container => {
                    view.Html.DevExpress().HyperLink(hyperlink => {
                        var keyValue = container.KeyValue;
                        var description = DataBinder.Eval(container.DataItem, "Description");

                        hyperlink.Name = "hl" + keyValue.ToString();
                        hyperlink.Properties.Text = description.ToString();
                        hyperlink.NavigateUrl = view.Url.Action("Details", "Home", new { key = keyValue });
                    }).Render();
                });
            });
        }

        return settings;
    }
}

File: VB/Models/GridViewSettingsHelper.vb
Imports System.Web.Mvc
Imports System.Web.UI
Imports DevExpress.Web.Mvc
Imports DevExpress.Web.Mvc.UI

Public Module GridViewSettingsHelper
	<System.Runtime.CompilerServices.Extension> _
	Public Function CreateGridViewSettings(ByVal view As WebViewPage, ByVal gridType As GridType) As GridViewSettings
		Dim settings As New GridViewSettings()
		settings.Name = "GridView" & gridType.ToString()
		settings.CallbackRouteValues = New With {Key .Controller = "Home", Key .Action = "GridViewPartial", Key .gridType = gridType}
		settings.KeyFieldName = "Id"

		If gridType = GridType.Invoices Then
			settings.Columns.Add("Id")
			settings.Columns.Add("Price")
			settings.Columns.Add(Sub(column)
				column.Caption = "Description"
				column.SetDataItemTemplateContent(Sub(container)
					view.Html.DevExpress().HyperLink(Sub(hyperlink)
						Dim keyValue = container.KeyValue
						Dim description = DataBinder.Eval(container.DataItem, "Description")
						hyperlink.Name = "hl" & keyValue.ToString()
						hyperlink.Properties.Text = description.ToString()
						hyperlink.NavigateUrl = view.Url.Action("Details", "Home", New With {Key .key = keyValue})
					End Sub).Render()
				End Sub)
			End Sub)
		End If

		Return settings
	End Function
End Module